terraform/lib/eks/eks.tf

```diff
@@ -0,0 +1,123 @@
+  #   So, by default the security groups are restrictive. Users needs to enable rules for specific ports required for App requirement or Add-ons
+  #   See the notes below for each rule used in these examples
+  #----------------------------------------------------------------------------------------------------------#
+  node_security_group_additional_rules = {
+    # Extend node-to-node security group rules. Recommended and required for the Add-ons
+    ingress_self_all = {
+      description = "Node to node all ports/protocols"
+      protocol    = "-1"
+      from_port   = 0
+      to_port     = 0
+      type        = "ingress"
+      self        = true
+    }
+    # Recommended outbound traffic for Node groups
+    egress_all = {
+      description      = "Node all egress"
+      protocol         = "-1"
+      from_port        = 0
+      to_port          = 0
+      type             = "egress"
+      cidr_blocks      = ["0.0.0.0/0"]
+      ipv6_cidr_blocks = ["::/0"]
+    }
+    # Allows Control Plane Nodes to talk to Worker nodes on all ports. Added this to simplify the example and further avoid issues with Add-ons communication with Control plane.
+    # This can be restricted further to specific port based on the requirement for each Add-on e.g., metrics-server 4443, spark-operator 8080, karpenter 8443 etc.
+    # Change this according to your security requirements if needed
+    ingress_cluster_to_node_all_traffic = {
+      description                   = "Cluster API to Nodegroup all traffic"
+      protocol                      = "-1"
+      from_port                     = 0
+      to_port                       = 0
+      type                          = "ingress"
+      source_cluster_security_group = true
+    }
+  }
+
+  managed_node_groups = {
+    mg_5 = {
+      node_group_name      = "managed-ondemand"
+      instance_types       = ["m5.large"]
+      subnet_ids           = var.subnet_ids
+      force_update_version = true
+    }
+  }
+
+  tags = var.tags
+}
+
+module "eks_cluster_kubernetes_addons" {
+  source = "github.com/aws-ia/terraform-aws-eks-blueprints//modules/kubernetes-addons?ref=v4.19.0"
+
+  eks_cluster_id               = module.eks_cluster.eks_cluster_id
+  eks_cluster_endpoint         = module.eks_cluster.eks_cluster_endpoint
+  eks_oidc_provider            = module.eks_cluster.oidc_provider
+  eks_cluster_version          = module.eks_cluster.eks_cluster_version
+  eks_worker_security_group_id = module.eks_cluster.worker_node_security_group_id
+  auto_scaling_group_names     = module.eks_cluster.self_managed_node_group_autoscaling_groups
+
+  # EKS Addons
+  enable_amazon_eks_vpc_cni = true
+  amazon_eks_vpc_cni_config = {
+    most_recent = true
+  }
+
+  enable_amazon_eks_coredns = true
+  amazon_eks_coredns_config = {
+    most_recent = true
+  }
+
+  enable_amazon_eks_kube_proxy         = true
+  enable_amazon_eks_aws_ebs_csi_driver = true
+
+  enable_prometheus                    = true
```

Comment:
I'd say lets remove Prometheus, AMP and Kyverno for now until we have a better idea what we want to use

Reply:
Prometheus, AMP and Kyverno are removed